### PR TITLE
Add button link pattern

### LINKED ISSUE #2642: No class to style a button as a link
Currently there is no pattern to style a button as a link. There is the [base button class](https://docs.vanillaframework.io/patterns/buttons/#base) which is good in many cases but sometimes the button needs to look and behave exactly like a link.

## Visual
Screenshot from 2019-11-13 09-46-56

## Context
In MAAS we are currently using a custom style to override the inherited button styles. This is also used in RBAC:

```
.p-button--link {
  @extend %p-button--base;

  border: 0;
  color: $color-link;
  line-height: 1.25rem;
  margin-bottom: 0;
  padding: 0;
  pointer-events: all;
  text-align: left;

  &:hover {
    background-color: inherit;
    text-decoration: underline;
  }

  &:visited {
    color: $color-link;
  }

  // Override inherited button style from Vanilla
  &:active {
    background-color: inherit !important;
  }
}
```

## State
Screenshot from 2019-11-13 09-47-40

Screenshot from 2019-11-13 09-47-22

### PR DESCRIPTION
**Pull request by sowasred2012**
## Done

- Added `p-button--link` variant
- This is the first new feature since the last release, so bumped version to 2.30.0

Fixes #2642 

## QA

- Open [demo](https://vanilla-framework-3724.demos.haus/docs/examples/patterns/buttons/link)
- See that the button looks identical to a [default link](https://vanilla-framework-3724.demos.haus/docs/patterns/links#default)
- Review updated documentation:
  - [Component status](https://vanilla-framework-3724.demos.haus/docs/component-status)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [component status page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/component-status.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

File: templates/docs/patterns/buttons.md [hunk 1]
```diff
@@ -56,6 +56,16 @@ You can use the brand button with the main color of your brand.
 View example of the brand button pattern
 </a></div>
 
+### Link
+
+<span class="p-label--new">New</span>
+
+In some contexts you may need a button to look visually identical a link.
```
Comment (lyubomir-popov) [inline templates/docs/patterns/buttons.md:63]:
```suggestion
In some contexts you may need a button to look visually identical to a link.
```

File: scss/_patterns_buttons.scss [hunk 2]
```diff
@@ -136,6 +137,33 @@ $hover-background-opacity-percentage: $hover-background-opacity-amount * 100%;
...
 }
 
+@mixin vf-button-link {
+  .p-button--link {
+    @extend %vf-link-base;
+
+    border: none;
+    border-radius: 0;
+    margin-right: 0;
+    margin-top: 0;
```
Comment (lyubomir-popov) [inline scss/_patterns_buttons.scss:147]: I believe his overrides another 0 - is it really necessary?

**Review by lyubomir-popov — APPROVED** [review]
LGTM, except for a typo - and a suggestion.

File: scss/_patterns_buttons.scss (hunk 2 above)
Comment (sowasred2012) [reply scss/_patterns_buttons.scss]: Ah yes - updated!

**Pull request merged**